OpenAI.SDK/Extensions/OpenAIServiceCollectionExtensions.cs

```diff
@@ -18,6 +20,11 @@ public static IHttpClientBuilder AddOpenAIService(this IServiceCollection servic
             optionsBuilder.BindConfiguration(OpenAiOptions.SettingKey);
         }
 
-        return services.AddHttpClient<IOpenAIService, OpenAIService>();
+        var serviceProvider = optionsBuilder.Services.BuildServiceProvider();
```

Comment:
@Swimburger a little help here, please.

Reply:
Can you do this?

```csharp
return services.AddHttpClient<IOpenAIService, OpenAIService>()
    .ConfigurePrimaryHttpMessageHandler(provider =>
    {
        var options = provider.GetRequiredService<OpenAiOptions>();
        return string.IsNullOrEmpty(options.HttpProxy) ? 
            null : 
            new HttpClientHandler {Proxy = new WebProxy(options.HttpProxy), UseProxy = true};
    });
```

I haven't test this, I'm mostly worried the returning `null` isn't allowed.

---

IMO, this doesn't need to be a feature because you're already returning the HTTP client builder which can be build upon to add the proxy functionality. I'd leave it up to the consumer to add it the way you have already documented in the readme file.

Reply:
Thanks, @Swimburger. I got an error while trying to get options. 
![image](https://user-images.githubusercontent.com/1096362/233203292-cbf11bbf-4f90-44d1-8a8f-6519b200fdcc.png)

I think it would be nice to make it easier to use httpProxy, as many people seem to be asking for it. However, I believe it may be more complex than the current solution proposed in the pull request. Therefore, I will postpone this pull request for now. Instead, I plan to make changes to the Options in the next version, which will allow for easier customization of the library. If possible, I may merge this pull request with the changes after finding a solution to the options issue.

Reply:
Sorry for not actually testing this, but I think we're close.
The issue is probably that I should've wrapped the options in `IOptions<OpenAiOptions>`.